src/sql/workbench/services/notebook/browser/models/notebookModel.ts

```diff
@@ -651,6 +654,7 @@ export class NotebookModel extends Disposable implements INotebookModel {
 					if (tailSource[0] === '\r\n' || tailSource[0] === '\n') {
 						newlinesBeforeTailCellContent = tailSource.splice(0, 1)[0];
 					}
+					tailCell.attachments = attachments;
```

Comment:
So you're duplicating the attachments in both the original and the new cell? Two questions

1. What about the third cell if there is one? (if you select some text in the middle of a cell and split it'll create 3 cells)
2. We shouldn't just be duplicating the attachments if they don't exist in the cell. An attachment should only exist in the cell it's referenced in (mostly important for embedded images). Do the "extra" attachments get cleaned up when we recreate the attachments on source changing?

Reply:
I made some changes. Now, I only add the attachment if it's referenced in the cell source and it covers the scenario when the cell splits in 3 parts.